src/openvic-simulation/interface/GFX.cpp

```diff
@@ -0,0 +1,98 @@
+#include "GFX.hpp"
+
+using namespace OpenVic;
+using namespace OpenVic::GFX;
+using namespace OpenVic::NodeTools;
+
+Font::Font(std::string_view new_identifier, colour_t new_colour, std::string_view new_fontname)
+	: HasIdentifierAndColour { new_identifier, new_colour, false, true }, fontname { new_fontname } {}
+
+node_callback_t Sprite::expect_sprite(callback_t<std::unique_ptr<Sprite>&&> callback) {
+	return expect_dictionary_keys(
+		"spriteType", ZERO_OR_MORE, _expect_instance<Sprite, TextureSprite>(callback),
+		"progressbartype", ZERO_OR_MORE, _expect_instance<Sprite, ProgressBar>(callback),
+		"PieChartType", ZERO_OR_MORE, _expect_instance<Sprite, PieChart>(callback),
+		"LineChartType", ZERO_OR_MORE, _expect_instance<Sprite, LineChart>(callback),
+		"textSpriteType", ZERO_OR_MORE, _expect_instance<Sprite, TextureSprite>(callback),
+		"maskedShieldType", ZERO_OR_MORE, _expect_instance<Sprite, MaskedFlag>(callback),
+		"corneredTileSpriteType", ZERO_OR_MORE, success_callback,
+		"tileSpriteType", ZERO_OR_MORE, success_callback,
+		"BarChartType", ZERO_OR_MORE, success_callback,
+		"scrollingSprite", ZERO_OR_MORE, success_callback
+	);
+}
+
+TextureSprite::TextureSprite() : texture_file {}, no_of_frames { NO_FRAMES } {}
+
+bool TextureSprite::_fill_key_map(key_map_t& key_map) {
+	bool ret = Sprite::_fill_key_map(key_map);
+	ret &= add_key_map_entries(key_map,
+		"texturefile", ZERO_OR_ONE, expect_string(assign_variable_callback_string(texture_file)),
+		"textureFile", ZERO_OR_ONE, expect_string(assign_variable_callback_string(texture_file)),
+		"noOfFrames", ZERO_OR_ONE, expect_uint(assign_variable_callback(no_of_frames)),
+
+		"norefcount", ZERO_OR_ONE, success_callback,
+		"effectFile", ZERO_OR_ONE, success_callback,
```

Comment:
```suggestion
		"effectFile", ZERO_OR_ONE, success_callback, // TODO: Implement GFX Effects
```
I can image implementing effects will be a lot of fun.

Reply:
Yeah I really hope the shader conversion is feasible, gotta get Spartan on the case